src/main/java/calculationEngine/entities/CeEntity.java

```diff
@@ -200,6 +201,16 @@ public void setHitPoints(int hitPoints) {
         this.hitPoints = hitPoints;
     }
 
+    public void dealDamage(int damage) {
+        this.hitPoints -= damage;
+        this.damages.add(damage);
+    }
+    public List<Integer> getDamages() {
+        List<Integer> tmp = new ArrayList<>(damages);
```

Comment:
why are you creating a temporary list here which is just a copy from the already existing list?

Reply:
How would you do it?

Reply:
I want to return the list and then clear it. This was the way I found. :-D